packages/components/src/widgets/DistWidget/DistributionsChart.tsx

```diff
@@ -22,65 +24,120 @@ import {
   drawCircle,
   drawCursorLines,
   drawVerticalLine,
-  primaryColor,
 } from "../../lib/draw/index.js";
 import { Point } from "../../lib/draw/types.js";
 import { useCanvas, useCanvasCursor } from "../../lib/hooks/index.js";
 import { DrawContext } from "../../lib/hooks/useCanvas.js";
-import {
-  adjustColorBrightness,
-  canvasClasses,
-  flattenResult,
-} from "../../lib/utility.js";
+import { canvasClasses, flattenResult } from "../../lib/utility.js";
 import { PlotTitle } from "../PlotWidget/PlotTitle.js";
-import { DistProvider, useVerticalLineValue } from "./DistProvider.js";
+import { DistProvider, useGetSelectedVerticalLine } from "./DistProvider.js";
 import { SummaryTable } from "./SummaryTable.js";
 import { adjustPdfHeightToScale } from "./utils.js";
 
+const distRadiusScalingFromHeight = d3
+  .scaleLinear()
+  .domain([10, 300]) // The potential height of the chart
+  .range([2, 5]) // The range of circle radiuses
+  .clamp(true);
+
 export type DistributionsChartProps = {
   plot: SqDistributionsPlot;
   environment: Env;
   height: number;
 };
 
+// We have a similar function in squiggle-lang, but it's not exported, and this function is simple enough.
+type DataPoint = {
+  x: number;
+  y: number;
+};
+function interpolateYAtX(
+  xValue: number,
+  continuousData: { x: number; y: number }[],
+  yScale: d3.ScaleContinuousNumeric<number, number>
+): number | null {
+  let pointBefore: DataPoint | null = null,
+    pointAfter: DataPoint | null = null;
+  for (const point of continuousData) {
```

Comment:
The trick with `xInterpolate` below is cool, but this loop bothers me, it's a linear scan (instead of binary search) over 1000 points (worst case with default settings).

Exposing this through `SqPointSet` would be better but I checked the relevant code and I see that it'll take some refactorings, so I don't think this is a blocker, can be improved later.

Reply:
Good point. I also looked into things, but came to a similar conclusion.